llvm/lib/DWARFLinkerParallel/DWARFLinkerCompileUnit.cpp

```diff
@@ -167,6 +167,21 @@ void CompileUnit::cleanupDataAfterClonning() {
   getOrigUnit().clear();
 }
 
+/// Make a best effort to guess the
+/// Xcode.app/Contents/Developer/Toolchains/ path from an SDK path.
+static SmallString<128> guessToolChainBaseDir(StringRef SysRoot) {
+  SmallString<128> Result;
+  // Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
+  StringRef Base = SysRoot;
+  Base = sys::path::parent_path(Base);
+  if (sys::path::filename(Base) != "SDKs")
+    return Result;
+  Result = Base;
+  Base = sys::path::parent_path(Base);
+  sys::path::append(Result, "Toolchains");
+  return Result;
+}
```

Comment:
I really wish we could avoid duplicating this. Out of scope for this patch, but I think the parallel dwarf linker should live under DWARFLinker so we can facilitate sharing files. (CC @avl-llvm)

Reply:
Agreed, we need a way to share things. I would suggest a bit another solution. Let`s have DWARFLinkerBase library. Which would be linked to DWARFLinker and DWARFLinkerParallel. That library will contain all shared code. It also potentially could be used by other part trying to link DWARF. f.e. BOLT does similar things. 

What do you think on creating such a common library?

Reply:
Sounds good, but I still think we should move all that in a `DWARFLinker` directory (i.e. on the filesystem) to avoid the proliferation of `DWARFLinker*` libs. The Transforms directory for example is good prior art for that.

Reply:
f.e. 
DWARFLinker/
DWARFLinker/Apple
DWARFLinker/Parallel

DWARFLinker would contain code which goes to DWARFLinkerBase library.
DWARFLinker/Apple would contain code for apple DWARFLinker and will use DWARFLinkerBase library.
DWARFLinker/Parallel would contain code for llvm DWARFLinker and will use DWARFLinkerBase library.

?